osquery/tables/system/windows/hvci_status.cpp

```diff
@@ -0,0 +1,87 @@
+#include <osquery/sql.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genHVCIStatus(QueryContext& context) {
+  Row r;
+  QueryData results;
+
+  const WmiRequest wmiSystemReq("SELECT * FROM Win32_DeviceGuard",
+                                (BSTR)L"ROOT\\MICROSOFT\\WINDOWS\\DEVICEGUARD");
+  const std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  if (wmiResults.empty()) {
+    LOG(WARNING) << "Error retreiving information from WMI.";
+    return results;
+  }
+  for (const auto& data : wmiResults) {
+    long vbsmethod;
+    long codepolicystatusmethod;
+    long umcipolicymethod;
+
+    data.GetString("Version", r["version"]);
+    data.GetString("InstanceIdentifier", r["instance_identifier"]);
+    data.GetLong("VirtualizationBasedSecurityStatus", vbsmethod);
+    data.GetLong("CodeIntegrityPolicyEnforcementStatus",
+                 codepolicystatusmethod);
+    data.GetLong("UsermodeCodeIntegrityPolicyEnforcementStatus",
+                 umcipolicymethod);
+
+    std::string vbsmethod_str;
+    std::map<long, std::string> vbsmethods;
+
+    std::string codepolicystatusmethod_str;
+    std::map<long, std::string> codepolicystatusmethods;
```

Comment:
Same note as `vbs_methods` for both this and the `umcipolicymethods`.

Reply:
Fixed.